Added documentation to lib_camera_v2 (#119)
xmos/lib_camera: doc/rst/06_contributing.rst

The following section focuses on contributing to the ISP components. 

Modifying or Adding an ISP component
------------------------------------

The library supports two types of ISP components: Line ISP and End-of-Frame (EOF) ISP. The primary difference between them lies in how they process image data.

Line ISP components process data on a line-by-line basis or in groups of lines. This means the component processes each line of the image as it is received from the camera sensor. This approach is typically used when converting from RAW8 input to a desired output format, such as RGB888 or YUV422. Processing line by line is preferred in this case to avoid storing the entire RAW image in memory.

End-of-Frame ISP components, on the other hand, process data after the entire image has been received and processed by the line ISP components. These components handle tasks that do not necessarily require storing the entire image, such as auto-exposure adjustments, or tasks that can be performed in-place in memory, such as int8 to uint8 conversion or gamma correction.

For modifying components, the user should be aware of the following:

> - camera_isp.c : this file contain both line process under the ``handle_expected_lines()`` function and EOF process under the ``handle_post_process()`` function. 

Reviewer: ```suggestion
- camera_isp.c : this file contains both line process under the ``handle_expected_lines()`` function and EOF process under the ``handle_post_process()`` function. 
```

- camera_isp.h : this file contains the different modes, and their corresponding maximum allowed regions for each mode. 

Other files that may be of interest are:

- camera.h : defines the absolute maximum sensor region. This will drive the maximum output image size.

- sensor_wrapper.h : Defines the sensor wrapper functions, which are used to communicate with the camera sensor and configure its settings. These wrapper functions are implemented in the ``sensor_wrapper.c`` file. This file serves as a C wrapper for the C++ implementation, specifically the ``sensor_imx219.cpp`` class, which is a subclass of the ``sensor_base.cpp`` class.

> In order to add or modify an ISP component the user would have to submit his changes in the `isp` folder. As an example, lets suppose user want to improve the conversion from RAW8 to RGB888 (RGB1) with an improved algorithm, or wants to add a new alogirthm, like RAW8 to YUV422.

Reviewer: ```suggestion
In order to add or modify an ISP component the user would have to submit their changes in the `isp` folder. As an example, lets suppose the user wants to improve the conversion from RAW8 to RGB888 (RGB1) with an improved algorithm, or wants to add a new alogrithm, like RAW8 to YUV422.
```
